### PR TITLE
Limit pods per run with k8s_job_executor

### PR DESCRIPTION
**Pull request by johannkm**
Duplicate of https://github.com/dagster-io/dagster/pull/7752?no-redirect=1 so that I [could push changes](https://github.com/github/feedback/discussions/5634).

Closes https://github.com/dagster-io/dagster/issues/6580

### CODE REVIEW

**Comment by johannkm** [comment]
@alangenfeld @rexledesma could you guys check out:

- the `max_concurrent` name for this config. It matches the multiproc executor, but could be confusing with global op limits in the future?
- the added test

File: python_modules/dagster/dagster_tests/execution_tests/engine_tests/test_step_delegating_executor.py [hunk 1]
```diff
@@ -220,6 +216,50 @@ def test_execute_intervals():
...
 
+@solid
+def slow_solid(_):
+    time.sleep(2)
+
+
+@pipeline(
+    mode_defs=[
+        ModeDefinition(
+            executor_defs=[test_step_delegating_executor],
```
Comment (johannkm) [inline python_modules/dagster/dagster_tests/execution_tests/engine_tests/test_step_delegating_executor.py:227]: unfortunately still hard to do tests with an in process run worker but non default executor when using Jobs

File: python_modules/libraries/dagster-k8s/dagster_k8s/executor.py [hunk 2]
```diff
@@ -35,6 +35,11 @@
         {
             "job_namespace": Field(StringSource, is_required=False),
             "retries": get_retries_config(),
+            "max_concurrency": Field(
+                IntSource,
+                is_required=False,
+                description="Limit on number of concurrent pods launched per run",
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-k8s/dagster_k8s/executor.py:41]: [1]

**Review by alangenfeld — COMMENTED** [review]
> the max_concurrent name for this config. It matches the multiproc executor, but could be confusing with global op limits in the future?

I think its fine, just update [1] with language to make it clear

File: python_modules/dagster/dagster_tests/execution_tests/engine_tests/test_step_delegating_executor.py (hunk 1 above)
Comment (johannkm) [reply python_modules/dagster/dagster_tests/execution_tests/engine_tests/test_step_delegating_executor.py]: you're right, not sure what I ran in to

File: python_modules/libraries/dagster-k8s/dagster_k8s/executor.py [hunk 3]
```diff
@@ -35,6 +35,12 @@
         {
             "job_namespace": Field(StringSource, is_required=False),
             "retries": get_retries_config(),
+            "max_concurrency": Field(
+                IntSource,
+                is_required=False,
+                description="Limit on number of concurrent pods launched per run. Note that this is "
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-k8s/dagster_k8s/executor.py:41]: nit: maybe
concurrent pods launched per run -> concurrent pods runnings within the scope of this run

**Review by alangenfeld — APPROVED** [review]
noice

File: python_modules/libraries/dagster-k8s/dagster_k8s/executor.py [hunk 4]
```diff
@@ -35,6 +35,12 @@
         {
             "job_namespace": Field(StringSource, is_required=False),
             "retries": get_retries_config(),
+            "max_concurrency": Field(
```
[the remark this answers is not in the document]
Comment (kervel) [reply python_modules/libraries/dagster-k8s/dagster_k8s/executor.py:38]: this should be max_concurrent.

**Review by kervel — COMMENTED** [review]
hello @johannkm , i was in the process of upgrading my dagster installation and i see that we didn't spot a typo ...

**Pull request merged**